src/EnergyPlus/SystemAvailabilityManager.cc

```diff
@@ -602,182 +582,162 @@ namespace SystemAvailabilityManager {
+                nightCycleMgr.FanSched = cAlphaArgs(3);
+                nightCycleMgr.FanSchedPtr = GetScheduleIndex(state, cAlphaArgs(3));
+                if (nightCycleMgr.FanSchedPtr == 0) {
+                    ShowSevereError(state, format("{}{}=\"{}\", invalid", RoutineName, cCurrentModuleObject, cAlphaArgs(1)));
                     ShowContinueError(state, "not found: " + cAlphaFieldNames(3) + "=\"" + cAlphaArgs(3) + "\".");
                     ErrorsFound = true;
                 }
 
-                {
-                    auto const SELECT_CASE_var(UtilityRoutines::MakeUPPERCase(cAlphaArgs(4)));
-                    if (SELECT_CASE_var == "STAYOFF") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = StayOff;
-                    } else if (SELECT_CASE_var == "CYCLEONANY") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = CycleOnAny;
-                    } else if (SELECT_CASE_var == "CYCLEONCONTROLZONE") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = CycleOnControlZone;
-                    } else if (SELECT_CASE_var == "CYCLEONANYZONEFANSONLY") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = ZoneFansOnly;
-                    } else if (SELECT_CASE_var == "CYCLEONANYCOOLINGORHEATINGZONE") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = CycleOnAnyCoolingOrHeatingZone;
-                    } else if (SELECT_CASE_var == "CYCLEONANYCOOLINGZONE") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = CycleOnAnyCoolingZone;
-                    } else if (SELECT_CASE_var == "CYCLEONANYHEATINGZONE") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = CycleOnAnyHeatingZone;
-                    } else if (SELECT_CASE_var == "CYCLEONANYHEATINGZONEFANSONLY") {
-                        state.dataSystemAvailabilityManager->NCycSysAvailMgrData(SysAvailNum).CtrlType = CycleOnAnyHeatingZoneFansOnly;
-                    } else {
-                        ShowSevereError(state, std::string{RoutineName} + cCurrentModuleObject + "=\"" + cAlphaArgs(1) + "\", invalid");
-                        ShowSevereError(state, std::string{RoutineName} + "incorrect value: " + cAlphaFieldNames(4) + "=\"" + cAlphaArgs(4) + "\".");
-                        ErrorsFound = true;
-                    }
-                }
+                nightCycleMgr.nightCycleControlType = static_cast<NightCycleControlType>(
+                    getEnumerationValue(NightCycleControlTypeNamesUC, UtilityRoutines::MakeUPPERCase(cAlphaArgs(4))));
+
+                assert(nightCycleMgr.nightCycleControlType != NightCycleControlType::Invalid);
```

Comment:
Is this right? Do we really want an `assert` here or should we be handling the error like happed in the deleted code?

Reply:
Yes. The epJSON parser catches illegal enumeration inputs and fatals out. We should not see Invalids from `getEnumerationValue` because the program would have already terminated.

Reply:
@mitchute, yes. The InputProcessor `getInput` functions have been reimplemented on top of epJSON. It's IDF -> epJSON -> InputProcessor. At some point in the near future, we will start stripping off the InputProcessor functions and initializing EnergyPlus objects directly from epJSON as is currently being done in `GroundHeatExchanger.cc` (I assumed you did that) and one or two other modules.

Reply:
Verified. A4 of `AvailabilityManager:NightCycle` is of type choice.

Reply:
@jmythms you probably already checked this, but I'll just note that the key choice fields in the IDD will need to match exactly the strings you have in your arrays for this to work. As long as those match, we're good.

Reply:
Verified, this and the other places where assert was used as well. :) Yeah, I usually create the arrays from [the strings for comparison](https://github.com/NREL/EnergyPlus/blob/32ed589540a2154d0e696bffb08b09f0fb9ba9d5/src/EnergyPlus/SystemAvailabilityManager.cc#L626-L649) (like "STAYOFF") that are already in the code. But yes, I understand the correct way to do it is to take the keys from the IDD. Will do it like that from now on.